src/frontend/context/ConfigContext.tsx

```diff
@@ -95,81 +95,87 @@ const defaultContext: ConfigContextType = [
 
 const ConfigContext = React.createContext<ConfigContextType>(defaultContext);
 
-type Props = {
-  children: React.ReactNode;
-};
-
-export const ConfigProvider = ({ children }: Props) => {
+export const ConfigProvider = ({ children }: React.PropsWithChildren<{}>) => {
+  const mountedRef = React.useRef(false);
```

Comment:
We use this is a few places, perhaps handy to extract to a custom hook e.g. https://www.npmjs.com/package/ismounted or our own - that we avoid the possibility of forgetting to set this in a `useEffect`. With putting it into a `useEffect` that does other things (e.g. the initial `loadConfig` effect below) there is always the risk that we refactor that resulting in the useEffect running after first mount, which would break this `isMounted`

Reply:
yeah turns out we had a custom hook already defined for this so updated it and used here (and in other areas that did a similar thing)